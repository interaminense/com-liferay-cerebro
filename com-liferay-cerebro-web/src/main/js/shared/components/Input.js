import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

import Button from './Button';

const SIZES = ['lg', 'sm'];

class InputAddon extends Component {
	render() {
		return <div class="input-group-addon">{this.props.children}</div>;
	}
}

class InputButton extends Component {
	render() {
		return (
			<span class="input-group-btn">
				<Button {...this.otherProps()}>{this.props.children}</Button>
			</span>
		);
	}
}

class InputInset extends Component {
	render() {
		return (
			<span class="input-group-inset-item">{this.props.children}</span>
		);
	}
}

class InputGroupInput extends Component {
	render() {
		return <div class="input-group-input">{this.props.children}</div>;
	}
}

class InputGroup extends Component {
	render() {
		const {children, inset, size} = this.props;

		const classes = getCN('input-group', {
			'input-group-inset': inset,
			[`input-group-${size}`]: size
		});

		return (
			<div class={classes} {...this.otherProps()}>
				{children}
			</div>
		);
	}
}

InputGroup.PROPS = {
	inset: Config.bool().value(false),
	size: Config.oneOf(SIZES)
};

class Input extends Component {
	/**
	 * Public method used for focusing the input element.
	 */
	focus() {
		this.element.focus();
	}

	/**
	 * Public method used for retrieving value of input.
	 */
	getValue() {
		return this.element.value;
	}

	render() {
		const {size} = this.props;

		const classes = getCN('input-root', 'form-control', {
			[`form-control-${size}`]: size
		});

		return <input type="text" {...this.otherProps()} class={classes} />;
	}
}

Input.PROPS = {
	size: Config.oneOf(SIZES)
};

Input.Addon = InputAddon;
Input.Group = InputGroup;
Input.GroupInput = InputGroupInput;
Input.Button = InputButton;
Input.Inset = InputInset;

Input.SIZES = SIZES;

export default Input;