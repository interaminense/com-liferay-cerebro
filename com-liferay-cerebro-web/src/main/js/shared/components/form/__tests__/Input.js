import Input from '../Input';

describe('Input', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Input({name: 'foo'});

		expect(component).toMatchSnapshot();
	});
});