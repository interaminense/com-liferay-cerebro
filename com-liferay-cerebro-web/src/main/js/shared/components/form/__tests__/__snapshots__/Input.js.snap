// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render 1`] = `
<div class="">
  <input id="FormInput1" value="" class="input-root form-control" type="text">
</div>
`;
